// react-youtube-lite/dist/react-youtube-lite.esm.js
import { useState, createElement, Fragment } from 'react';

function styleInject(css, ref) {
  if ( ref === void 0 ) ref = {};
  var insertAt = ref.insertAt;

  if (!css || typeof document === 'undefined') { return; }

  var head = document.head || document.getElementsByTagName('head')[0];
  var style = document.createElement('style');
  style.type = 'text/css';

  if (insertAt === 'top') {
    if (head.firstChild) {
      head.insertBefore(style, head.firstChild);
    } else {
      head.appendChild(style);
    }
  } else {
    head.appendChild(style);
  }

  if (style.styleSheet) {
    style.styleSheet.cssText = css;
  } else {
    style.appendChild(document.createTextNode(css));
  }
}

var css_248z = ".embed-responsive{position:relative!important;display:block!important;height:0!important;padding:0!important;overflow:hidden!important}.embed-responsive .embed-responsive-item,.embed-responsive embed,.embed-responsive iframe,.embed-responsive object,.embed-responsive video{position:absolute;top:0;left:0;bottom:0;height:100%;width:100%;border:0}.aspect-ratio-none{padding-bottom:0!important}.aspect-ratio-square{padding-bottom:100%!important}.aspect-ratio-16\\/9{padding-bottom:56.25%!important}.aspect-ratio-4\\/3{padding-bottom:75%!important}.aspect-ratio-21\\/9{padding-bottom:42.86%!important}.ryt-lite{background-color:#000;position:relative;display:block;contain:content;background-position:50%;background-size:cover;cursor:pointer}.ryt-lite:before{content:\"\";display:block;position:absolute;top:0;background-image:url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAADGCAYAAAAT+OqFAAAAdklEQVQoz42QQQ7AIAgEF/T/D+kbq/RWAlnQyyazA4aoAB4FsBSA/bFjuF1EOL7VbrIrBuusmrt4ZZORfb6ehbWdnRHEIiITaEUKa5EJqUakRSaEYBJSCY2dEstQY7AuxahwXFrvZmWl2rh4JZ07z9dLtesfNj5q0FU3A5ObbwAAAABJRU5ErkJggg==);background-position:top;background-repeat:repeat-x;height:60px;padding-bottom:50px;width:100%;transition:all .2s cubic-bezier(0,0,.2,1)}.ryt-lite:after{content:\"\";display:block;padding-bottom:56.25%}.ryt-lite>iframe{width:100%;height:100%;position:absolute;top:0;left:0}.ryt-lite>.lty-playbtn{width:70px;height:46px;background-color:#212121;z-index:1;opacity:.8;border-radius:14%;transition:all .2s cubic-bezier(0,0,.2,1)}.ryt-lite:hover>.lty-playbtn{background-color:red;opacity:1}.ryt-lite>.lty-playbtn:before{content:\"\";border-color:transparent transparent transparent #fff;border-style:solid;border-width:11px 0 11px 19px}.ryt-lite>.lty-playbtn,.ryt-lite>.lty-playbtn:before{position:absolute;top:50%;left:50%;transform:translate3d(-50%,-50%,0)}.ryt-lite.lyt-activated{cursor:unset}.ryt-lite.lyt-activated:before,.ryt-lite.lyt-activated>.lty-playbtn{opacity:0;pointer-events:none}";
styleInject(css_248z);

// Dependencies
/**
 * Returns the ID of the youtube video url provided
 *
 * @param url - The URL of the video (we take care of )
 */

function getYoutubeId(url) {
  var regExp = /^.*((youtu.be\/)|(v\/)|(\/u\/\w\/)|(embed\/)|(watch\?))\??v?=?([^#&?]*).*/;
  var match = url.match(regExp);
  return match && match[7].length == 11 ? match[7] : '';
}
/**
 * Render faster YouTube videos to get a high score page render
 */

var ReactYouTubeLite = function ReactYouTubeLite(_ref) {
  var _ref$adNetwork = _ref.adNetwork,
      adNetwork = _ref$adNetwork === void 0 ? true : _ref$adNetwork,
      _ref$url = _ref.url,
      url = _ref$url === void 0 ? '' : _ref$url,
      _ref$playlist = _ref.playlist,
      playlist = _ref$playlist === void 0 ? false : _ref$playlist,
      _ref$poster = _ref.poster,
      poster = _ref$poster === void 0 ? 'hqdefault' : _ref$poster,
      _ref$title = _ref.title,
      title = _ref$title === void 0 ? 'React YouTube Lite' : _ref$title,
      _ref$noCookie = _ref.noCookie,
      noCookie = _ref$noCookie === void 0 ? false : _ref$noCookie,
      _ref$activatedClass = _ref.activatedClass,
      activatedClass = _ref$activatedClass === void 0 ? 'lyt-activated' : _ref$activatedClass,
      _ref$iframeClass = _ref.iframeClass,
      iframeClass = _ref$iframeClass === void 0 ? 'embed-responsive-item' : _ref$iframeClass,
      _ref$playerClass = _ref.playerClass,
      playerClass = _ref$playerClass === void 0 ? 'lty-playbtn' : _ref$playerClass,
      _ref$wrapperClass = _ref.wrapperClass,
      wrapperClass = _ref$wrapperClass === void 0 ? 'ryt-lite embed-responsive' : _ref$wrapperClass,
      _ref$aspectRatio = _ref.aspectRatio,
      aspectRatio = _ref$aspectRatio === void 0 ? 'aspect-ratio-16/9' : _ref$aspectRatio;

  var _React$useState = useState(false),
      preconnected = _React$useState[0],
      setPreconnected = _React$useState[1];

  var _React$useState2 = useState(false),
      iframe = _React$useState2[0],
      setIframe = _React$useState2[1];

  var videoId = encodeURIComponent(getYoutubeId(url));
  var videoTitle = title;
  var posterUrl = "https://i.ytimg.com/vi/" + videoId + "/" + poster + ".jpg";
  var ytUrl = noCookie ? 'https://www.youtube-nocookie.com' : 'https://www.youtube.com';
  var iframeSrc = !playlist ? ytUrl + "/embed/" + videoId + "?autoplay=1" : ytUrl + "/embed/videoseries?list=" + videoId;
  /**
   * Preload google `link` in header
   */

  var warmConnections = function warmConnections() {
    if (preconnected) return;
    setPreconnected(true);
  };
  /**
   * Set iframe as active if already active, return.
   */


  var addIframe = function addIframe() {
    if (iframe) return;
    setIframe(true);
  };

  return createElement(Fragment, null, createElement("link", {
    rel: "preload",
    href: posterUrl,
    as: "image"
  }), createElement(Fragment, null, preconnected && createElement(Fragment, null, createElement("link", {
    rel: "preconnect",
    href: ytUrl
  }), createElement("link", {
    rel: "preconnect",
    href: "https://www.google.com"
  }), adNetwork && createElement(Fragment, null, createElement("link", {
    rel: "preconnect",
    href: "https://static.doubleclick.net"
  }), createElement("link", {
    rel: "preconnect",
    href: "https://googleads.g.doubleclick.net"
  })))), createElement("div", {
    onPointerOver: warmConnections,
    onClick: addIframe,
    className: wrapperClass + " " + aspectRatio + " " + (iframe ? activatedClass : ''),
    "data-title": videoTitle,
    style: {
      backgroundImage: "url(" + posterUrl + ")"
    }
  }, createElement("div", {
    className: playerClass
  }), iframe && createElement("iframe", {
    className: iframeClass,
    title: videoTitle,
    width: "560",
    height: "315",
    frameBorder: "0",
    allow: "accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture",
    allowFullScreen: true,
    src: iframeSrc
  })));
};

export { ReactYouTubeLite, getYoutubeId };
