// react-youtube-lite/dist/react-youtube-lite.cjs.development.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');

// Dependencies
/**
 * Returns the ID of the youtube video url provided
 *
 * @param url - The URL of the video (we take care of )
 */

function getYoutubeId(url) {
  var regExp = /^.*((youtu.be\/)|(v\/)|(\/u\/\w\/)|(embed\/)|(watch\?))\??v?=?([^#&?]*).*/;
  var match = url.match(regExp);
  return match && match[7].length == 11 ? match[7] : '';
}
/**
 * Render faster YouTube videos to get a high score page render
 */

var ReactYouTubeLite = function ReactYouTubeLite(_ref) {
  var _ref$adNetwork = _ref.adNetwork,
      adNetwork = _ref$adNetwork === void 0 ? true : _ref$adNetwork,
      _ref$url = _ref.url,
      url = _ref$url === void 0 ? '' : _ref$url,
      _ref$playlist = _ref.playlist,
      playlist = _ref$playlist === void 0 ? false : _ref$playlist,
      _ref$poster = _ref.poster,
      poster = _ref$poster === void 0 ? 'hqdefault' : _ref$poster,
      _ref$title = _ref.title,
      title = _ref$title === void 0 ? 'React YouTube Lite' : _ref$title,
      _ref$noCookie = _ref.noCookie,
      noCookie = _ref$noCookie === void 0 ? false : _ref$noCookie,
      _ref$activatedClass = _ref.activatedClass,
      activatedClass = _ref$activatedClass === void 0 ? 'lyt-activated' : _ref$activatedClass,
      _ref$iframeClass = _ref.iframeClass,
      iframeClass = _ref$iframeClass === void 0 ? 'embed-responsive-item' : _ref$iframeClass,
      _ref$playerClass = _ref.playerClass,
      playerClass = _ref$playerClass === void 0 ? 'lty-playbtn' : _ref$playerClass,
      _ref$wrapperClass = _ref.wrapperClass,
      wrapperClass = _ref$wrapperClass === void 0 ? 'ryt-lite embed-responsive' : _ref$wrapperClass,
      _ref$aspectRatio = _ref.aspectRatio,
      aspectRatio = _ref$aspectRatio === void 0 ? 'aspect-ratio-16/9' : _ref$aspectRatio;

  var _React$useState = React.useState(false),
      preconnected = _React$useState[0],
      setPreconnected = _React$useState[1];

  var _React$useState2 = React.useState(false),
      iframe = _React$useState2[0],
      setIframe = _React$useState2[1];

  var videoId = encodeURIComponent(getYoutubeId(url));
  var videoTitle = title;
  var posterUrl = "https://i.ytimg.com/vi/" + videoId + "/" + poster + ".jpg";
  var ytUrl = noCookie ? 'https://www.youtube-nocookie.com' : 'https://www.youtube.com';
  var iframeSrc = !playlist ? ytUrl + "/embed/" + videoId + "?autoplay=1" : ytUrl + "/embed/videoseries?list=" + videoId;
  /**
   * Preload google `link` in header
   */

  var warmConnections = function warmConnections() {
    if (preconnected) return;
    setPreconnected(true);
  };
  /**
   * Set iframe as active if already active, return.
   */


  var addIframe = function addIframe() {
    if (iframe) return;
    setIframe(true);
  };

  return React.createElement(React.Fragment, null, React.createElement("link", {
    rel: "preload",
    href: posterUrl,
    as: "image"
  }), React.createElement(React.Fragment, null, preconnected && React.createElement(React.Fragment, null, React.createElement("link", {
    rel: "preconnect",
    href: ytUrl
  }), React.createElement("link", {
    rel: "preconnect",
    href: "https://www.google.com"
  }), adNetwork && React.createElement(React.Fragment, null, React.createElement("link", {
    rel: "preconnect",
    href: "https://static.doubleclick.net"
  }), React.createElement("link", {
    rel: "preconnect",
    href: "https://googleads.g.doubleclick.net"
  })))), React.createElement("div", {
    onPointerOver: warmConnections,
    onClick: addIframe,
    className: wrapperClass + " " + aspectRatio + " " + (iframe ? activatedClass : ''),
    "data-title": videoTitle,
    style: {
      backgroundImage: "url(" + posterUrl + ")"
    }
  }, React.createElement("div", {
    className: playerClass
  }), iframe && React.createElement("iframe", {
    className: iframeClass,
    title: videoTitle,
    width: "560",
    height: "315",
    frameBorder: "0",
    allow: "accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture",
    allowFullScreen: true,
    src: iframeSrc
  })));
};

exports.ReactYouTubeLite = ReactYouTubeLite;
exports.getYoutubeId = getYoutubeId;
